company.el

```diff
@@ -645,16 +645,35 @@ commands in the `company-' namespace, abort completion."
                         (repeat :tag "Commands" function))
                  (repeat :tag "Commands" function)))
 
-(defcustom company-show-numbers nil
-  "If enabled, show quick-access numbers for the first ten candidates."
+(define-obsolete-variable-alias
+ 'company-show-numbers
+ 'company-show-quick-access
+ "0.10.0")
+
+(defcustom company-show-quick-access nil
+  "Show a quick-access column for the first ten candidates.
+
+Non-nil value enables a quick-access column on the right side of the tooltip,
+ unless the configured value is \\='left.
+If the value is nil, do not show a quick-access column.
+NOTE: A value of nil does not disable `M-(digit)' bindings.
+
+To customize shown quick-access keys, use `company-quick-access-key-producer'."
   :type '(choice (const :tag "off" nil)
                  (const :tag "left" 'left)
                  (const :tag "on" 't)))
 
-(defcustom company-show-numbers-function #'company--show-numbers
-  "Function called to get quick-access numbers for the first ten candidates.
+(define-obsolete-variable-alias
+ 'company-show-numbers-function
+ 'company-quick-access-key-producer
+ "0.10.0")
+
+(defcustom company-quick-access-key-producer #'company-quick-access-digit
```

Comment:
Call this one `company-quick-access-key-formatter`, maybe?

Reply:
Good idea!